clients/src/main/java/org/apache/kafka/common/requests/OffsetCommitResponse.java

```diff
@@ -149,13 +161,15 @@ public Builder addPartition(
             return this;
         }
 
-        public <P> Builder addPartitions(
-            String topicName,
-            List<P> partitions,
-            Function<P, Integer> partitionIndex,
-            Errors error
+        public final <P> Builder<T> addPartitions(
```

Comment:
There is still a problem here if `topicName` and `topicId` are both undefined in which case we should do what was done before and add to the response without caching.